File: frontend/src/app/models/doctor.ts
export interface DoctorUser {
  id: string;
  fullName: string;
  email: string;
}

export interface Doctor {
  id: number;
  applicationUserId: string;
  applicationUser?: DoctorUser;
  specialization: string;
  phoneNumber: string;
  availability: string;
  notes: string;
  createdAt: string;
}

export interface CreateDoctor {
  applicationUserId: string;
  specialization: string;
  phoneNumber: string;
  availability: string;
  notes: string;
}

File: backend/Models/Doctor.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Doctor
    {
        public int Id { get; set; }

        [Required]
        public string ApplicationUserId { get; set; } = string.Empty;

        public ApplicationUser? ApplicationUser { get; set; }

        [Required]
        public string Specialization { get; set; } = string.Empty;

        public string PhoneNumber { get; set; } = string.Empty;

        public string Availability { get; set; } = string.Empty;

        public string Notes { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}